tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/Program.cs

```diff
@@ -318,10 +316,41 @@ private static async Task SendDownstreamResponse(UpstreamResponse upstreamRespon
                 response.Headers.Add(header.Key, header.Value);
             }
 
-            var count = contentBytes ?? upstreamResponse.Content.Length;
+            long count = contentBytes ?? long.MaxValue;
+
             Log($"Writing response body of {count} bytes...");
-            await response.Body.WriteAsync(upstreamResponse.Content, 0, count);
-            Log($"Finished writing response body");
+
+            using Stream source = await upstreamResponse.GetContentStreamAsync(cancellationToken);
+            byte[] buffer = new byte[8192];
```

Comment:
```suggestion
            byte[] buffer = ArrayPool<byte>.Shared.Rent(81920);
```

I believe `ArrayPool<byte>.Shared.Rent()` should have better perf than `new byte[]`, due to less pressure on GC.  I also increased the buffer size to match what is used in `System.IO.Stream.CopyToAsync()`, which I believe should have slightly better perf:

https://github.com/dotnet/runtime/blob/313ec077c3975e5ff58c08ead212afc7fb58a24e/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L128

Reply:
thanks!

Reply:
Another option I'm thinking about (which can also happen in a separate PR).  If we create a class like `TruncatedStream` which wraps an inner stream but simulates truncating the length, we could then directly use `Stream.CopyToAsync()`, without needing to implement the byte copying ourselves.